### PR TITLE
Feature radio button refactor chris coleman

### PR DESCRIPTION
**Pull request by chrstphrclmn**
Backend changes required for multiple format searching.  Note, needs to be made live at the same time as the Radio Button Refactor branch for the frontend.

Also updated tests to run with new filter requirements.

### CODE REVIEW

File: Servers/Pivotal tc Server Developer Edition v4.0-config/catalina.policy [hunk 1]
```diff
@@ -0,0 +1,257 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (carolynrehm) [inline Servers/Pivotal tc Server Developer Edition v4.0-config/catalina.policy]: please remove Sever files

File: src/main/java/com/revature/controllers/ContentController.java [hunk 2]
```diff
@@ -97,19 +97,27 @@
...
 			moduleIdsList.add(Integer.parseInt(st.nextToken()));
 		}
+		
+		StringTokenizer ft = new StringTokenizer(format, ",");
+		while (ft.hasMoreTokens()) {
+			formatList.add(ft.nextToken());
+		}
 
-		Set<Content> contentSet = searchService.filter(title, format, moduleIdsList);
+		Set<Content> contentSet = searchService.filter(title, formatList, moduleIdsList);
```
Comment (carolynrehm) [inline src/main/java/com/revature/controllers/ContentController.java]: Why use a tokenizer here rather than just map the Request Param into an ArrayList?

Comment (carolynrehm) [reply src/main/java/com/revature/controllers/ContentController.java]: Also applicable for RequestController

Comment (chrstphrclmn) [reply src/main/java/com/revature/controllers/ContentController.java]: To maintain the convention used by previous team, they used tokenizer for Module

Comment (chrstphrclmn) [reply src/main/java/com/revature/controllers/ContentController.java]: As for why they chose to do that, I can't answer.

Comment (carolynrehm) [reply src/main/java/com/revature/controllers/ContentController.java]: I would refactor this; spring will do it for you

Comment (chrstphrclmn) [reply src/main/java/com/revature/controllers/ContentController.java]: Will do

Comment (chrstphrclmn) [reply src/main/java/com/revature/controllers/ContentController.java]: So I looked into this and the way the application is currently built, spring will not do it automatically in @RequestParam()

In order for RequestParam to catch an array of objects, the format of the request has to group the array like this.

param=value1&param=value2&param=value3

while our application currently passes arrays/lists like so

param=value1,value2,value3

That being said, the need for tokenizer is still not required as we can simply use the String.split(",") method to get our array.  I can of course change both ends to change the structure of how our request params are passed, but I think the split method would entail less gutting.  If your ok with it, that's the refactor I'd like to do.

Comment (chrstphrclmn) [reply src/main/java/com/revature/controllers/ContentController.java]: Nevermind, I found the documentation that allows for it in requestparam.
I'm an idiot.

File: src/main/java/com/revature/controllers/ContentController.java [hunk 3]
```diff
@@ -103,21 +103,12 @@
...
+			@RequestParam(value = "modules", required = false) List<Integer> modules) {
 		
-		StringTokenizer ft = new StringTokenizer(format, ",");
-		while (ft.hasMoreTokens()) {
-			formatList.add(ft.nextToken());
-		}
-
-		Set<Content> contentSet = searchService.filter(title, formatList, moduleIdsList);
+		System.out.println(formats.toString());
+		System.out.println(modules.toString());
```
Comment (carolynrehm) [inline src/main/java/com/revature/controllers/ContentController.java]: remove or replace with logger

Comment (carolynrehm) [reply src/main/java/com/revature/controllers/ContentController.java]: +SearchServiceImpl.java

Comment (chrstphrclmn) [reply src/main/java/com/revature/controllers/ContentController.java]: Sorry Carolyn, didnt mean for you to see this, these are for troubleshooting and will be removed.  I was making sure the spring refactor was working and noticed faulty behavior in the module search.  Currently diving into it more but will let you know after the investigation is done.

Comment (carolynrehm) [reply src/main/java/com/revature/controllers/ContentController.java]: No worries! Just ping me when it's ready for review

File: Servers/Pivotal tc Server Developer Edition v4.0-config/catalina.policy (hunk 1 above)
Comment (chrstphrclmn) [reply Servers/Pivotal tc Server Developer Edition v4.0-config/catalina.policy]: Removed

File: src/main/java/com/revature/controllers/ContentController.java (hunk 2 above)
Comment (chrstphrclmn) [reply src/main/java/com/revature/controllers/ContentController.java]: Fixed and pushed.

File: src/main/java/com/revature/controllers/ContentController.java (hunk 3 above)
Comment (chrstphrclmn) [reply src/main/java/com/revature/controllers/ContentController.java]: Sysouts have been removed.

**Review by carolynrehm — CHANGES_REQUESTED** [review]
Things like pom and curriculum files should be up to date with the dev branch. There are also files (e.g. src/main/java/com/revature/cmsforce/CMSforceApplication.java that don't have significant changes and should be removed from the PR) The PR should only include the files that are necessary

**Pull request closed without merging**